app/uk/gov/hmrc/transitmovements/repositories/DeparturesRepository.scala

```diff
@@ -62,15 +63,28 @@ import scala.util.control.NonFatal
 @ImplementedBy(classOf[DeparturesRepositoryImpl])
 trait DeparturesRepository {
   def insert(departure: Departure): EitherT[Future, MongoError, Unit]
-  def getDepartureWithoutMessages(eoriNumber: EORINumber, departureId: DepartureId): EitherT[Future, MongoError, Option[DepartureWithoutMessages]]
-  def getSingleMessage(eoriNumber: EORINumber, departureId: DepartureId, messageId: MessageId): EitherT[Future, MongoError, Option[MessageResponse]]
+
+  def getDepartureWithoutMessages(
+    eoriNumber: EORINumber,
+    departureId: DepartureId,
+    movementType: MovementType
+  ): EitherT[Future, MongoError, Option[DepartureWithoutMessages]]
+
+  def getSingleMessage(
+    eoriNumber: EORINumber,
+    departureId: DepartureId,
+    messageId: MessageId,
+    movementType: MovementType
+  ): EitherT[Future, MongoError, Option[MessageResponse]]
 
   def getMessages(
     eoriNumber: EORINumber,
     departureId: DepartureId,
+    movementType: MovementType,
     received: Option[OffsetDateTime]
   ): EitherT[Future, MongoError, Option[NonEmptyList[MessageResponse]]]
-  def getDepartures(eoriNumber: EORINumber): EitherT[Future, MongoError, Option[NonEmptyList[DepartureWithoutMessages]]]
+
+  def getDepartures(eoriNumber: EORINumber, movementType: MovementType): EitherT[Future, MongoError, Option[NonEmptyList[DepartureWithoutMessages]]]
```

Comment:
This doesn't need the `MovementType` here, because it's always going to be a `Departure`. Instead, in the implementation of the message, we just hardcode the filter to use `MovementType.Departure`.